demo-app/src/main.rs

```diff
@@ -0,0 +1,253 @@
+use std::marker::PhantomData;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use example_election::Election;
+// Items that should go in prelude
+use jmt::SimpleHasher;
+use sov_modules_api::{
+    mocks::{MockContext, MockPublicKey},
+    Context, DispatchCall, DispatchQuery, Genesis, Module, Spec,
+};
+use sov_modules_macros::{DispatchCall, DispatchQuery, Genesis, MessageCodec};
+use sov_state::{JmtStorage, Storage};
+use sovereign_sdk::{
+    core::{
+        mocks::MockProof,
+        traits::{BatchTrait, CanonicalHash, TransactionTrait},
+    },
+    jmt,
+    serial::Decode,
+    stf::{ConsensusSetUpdate, OpaqueAddress, StateTransitionFunction},
+};
+
+// we should re-export anyhow from the modules_api
+
+#[derive(Debug, PartialEq, Eq, Clone, BorshDeserialize, BorshSerialize)]
+pub struct DemoTransaction<C: Context> {
+    mock_signature: MockSignature,
+    msg: Vec<u8>,
+    phantom_context: PhantomData<C>,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, BorshDeserialize, BorshSerialize)]
+pub struct MockSignature {
+    pub_key: MockPublicKey,
+    msg_hash: [u8; 32],
+}
```

Comment:
non-blocking: 
1. We should use `sov-modules-api::mocks::MockSignature` instead, let's add TODO
2. Should we keep the public key outside of signature object?

Reply:
1. TODO added!

2. For a real signature scheme, the `pub_key` is either part of the signature or recovered from it using EcRecover, right? 
Maybe we should plan to use EcRecover and introduce separate types for unchecked and verified signatures. Maybe something like :

```rust 

struct RawSignature(Vec<u8>);
struct VerifiedSignature { 
   pub_key: PublicKey,
   msg_hash: [u8;32],
}
```

Reply:
But then we are limited to schemas that support pubkey recovery. Can we just send (PubKey, Sig) pair and verify the sig against the key?

Reply:
Oh, sure! I guess what I meant was that (once a signature is verified) we always have access to the pubkey. So we probably want that to be in the `VerifiedSignature` struct. It's fine if the pubkey is also included in the RawSignature type! 

The suggestion was really just adding a type boundary between Raw and Verified signatures